### PR TITLE
XI-58842 - initial set-up for the ng-masterdata repo

### PR DESCRIPTION
**Pull request by rkumar014**

### CODE REVIEW

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.routing.module.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { MasterdataComponent } from './masterdata.component';
+
+
+const routes: Routes = [
+  {
+    path: '',
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.routing.module.ts:8]: Here we will have to give the path for vendor component.

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/service/services/base.service.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  ) {
+    this.baseUrl = this.appSettings.roServiceUrl;
+    this.setUrls();
+  }
+
+  /**
+  * Method to set the api base url's
+  */
+  private setUrls(): void {
+
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/service/services/base.service.ts:27]: Set the url for workorder API here

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.module.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import { XpoimMasterdataComponent } from './xpoim-masterdata.component';
+import { MasterdataComponent } from './masterdata/masterdata.component';
+import { VendorComponent } from './masterdata/vendor/vendor.component';
+
+
+
+@NgModule({
+  declarations: [
+    XpoimMasterdataComponent,
+    MasterdataComponent,
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.module.ts:11]: Do not import the components here. Create a routing module and add it in module. In the routing module, based on the path load the module.

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.service.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { Injectable } from '@angular/core';
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.service.ts:1]: This file is not needed

File: intermodal-ng-masterdata/src/app/app.component.html [hunk 5]
```diff
@@ -0,0 +1,520 @@
+<!-- * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * -->
+<!-- * * * * * * * * * * * The content below * * * * * * * * * * * -->
+<!-- * * * * * * * * * * is only a placeholder * * * * * * * * * * -->
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/src/app/app.component.html:3]: Clear the content of this file

File: intermodal-ng-masterdata/projects/xpoim-masterdata/package.json [hunk 6]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "xpoim-masterdata",
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/package.json:2]: Please check if some commands need to be added

File: intermodal-ng-masterdata/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,36 @@
+/* To learn more about this file see: https://angular.io/config/tsconfig. */
+{
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/tsconfig.json:2]: check for prefix

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.component.html [hunk 8]
```diff
@@ -0,0 +1 @@
+<p>masterdata works!</p>
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.component.html:1]: Please remove this

**Review by xvvijay — CHANGES_REQUESTED** [review]
Please see inline comments

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.routing.module.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    component: MasterdataComponent
+  },
+  {
+    path: 'vendorCompliance',
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/masterdata/masterdata.routing.module.ts:12]: route should be like vendor-compliance

File: intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.module.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import { MasterdataComponent } from './masterdata/masterdata.component';
+import { VendorComplianceComponent } from './masterdata/vendor-compliance/vendor-compliance.component';
+
+
+
+@NgModule({
+  declarations: [
+  
+    MasterdataComponent,
+       VendorComplianceComponent
```
Comment (xvnambiar) [inline intermodal-ng-masterdata/projects/xpoim-masterdata/src/lib/xpoim-masterdata.module.ts:12]: vendor compliancecomponent, no need to add here.it has to be lazy loaded

**Review by xvnambiar — APPROVED** [review]

**Review by xvvijay — APPROVED** [review]

**Pull request merged**